File: src/main/java/xyz/ivyxjc/pubg4j/entity/PubgRoster.java
package xyz.ivyxjc.pubg4j.entity;

import java.util.ArrayList;
import java.util.List;
import lombok.Data;
import lombok.EqualsAndHashCode;
import xyz.ivyxjc.pubg4j.types.PlatformRegion;

/**
 * @author Ivyxjc
 * @since 4/22/2018
 */
@Data
@EqualsAndHashCode(of = {"rosterId"})
public class PubgRoster {
    private String matchId;
    private String rosterId;
    private PlatformRegion shardId;
    private Integer rank;
    private Integer teamId;
    private Boolean win;
    private List<PubgParticipant> participantList;

    public PubgRoster() {
        participantList = new ArrayList<>();
    }
}

File: src/main/kotlin/xyz/ivyxjc/pubg4j/entity/PubgRoster.kt
package xyz.ivyxjc.pubg4j.entity

import xyz.ivyxjc.pubg4j.types.PlatformRegion
import java.util.*

/**
 * @author Ivyxjc
 * @since 4/22/2018
 */
class PubgRoster : AbstractEntity {

    constructor()

    constructor(rosterId: String) {
        this.rosterId = rosterId
    }

    lateinit var rosterId: String

    var matchId: String? = null
    var shardId: PlatformRegion? = null
    var rank: Int? = null
    var teamId: Int? = null
    var win: Boolean? = null
    val participantList: List<PubgParticipant>

    init {
        participantList = ArrayList()
    }
}
